src/seals/util.py

```diff
@@ -23,6 +23,47 @@ def step(self, action):
         return obs, rew, False, info
 
 
+class MaskScoreWrapper(gym.Wrapper):
+    """Mask a list of box-shaped regions in the observation to hide reward info.
+
+    Intended for environments whose observations are raw pixels (like atari
+    environments). Used to mask regions of the observation that include information
+    that could be used to infer the reward, like the game score or enemy ship count.
+    """
+
+    def __init__(
+        self,
+        env: gym.Env,
+        score_regions: List[Dict[str, int]],
+        fill_value: Union[float, Sequence[float]] = 0,
+    ):
+        """Builds MaskScoreWrapper.
+
+        Args:
+            env: The environment to wrap.
+            score_regions: A list of box-shaped regions to mask, each denoted by
+                a dictionary `{"x0": x0, "x1": x1, "y0": y0, "y1": y1}`.
+            fill_value: The fill_value for the masked region. By default is black.
+                Can support RGB colors by being a sequence of values [r, g, b].
+        """
+        super().__init__(env)
+        self.fill_value = np.array(fill_value, env.observation_space.dtype)
```

Comment:
So the way the code is written, either works for RGB images, but I can add a test for it. If you provide one value, it will fill all channels with that value and if you provide a sequence it will fill the channels with the sequence. I did this because I wanted people to be able to mask with non-monochrome colors if they wanted to.

Reply:
Okay, so you're just relying on broadcasting, and assuming that the user will pass a sensible input that numpy will be able to broadcast. That seems fine